interpreter/runtime/data.ml

```diff
@@ -1,7 +1,15 @@
 type data = string ref
 type t = data
 
+exception Bounds
+
 let alloc bs = ref bs
+
 let size seg = I64.of_int_u (String.length !seg)
-let load seg i = (!seg).[Int64.to_int i]
+
+let load seg i =
+  let i = Int64.to_int i in
```

Comment:
Can we avoid shadowing variables like this?

Reply:
I usually like this style as it minimizes the number of variables in scope and fits easier in my head. I renamed to `i'`.